ui/src/containers/deployments/EditDeployment.js

```diff
@@ -149,23 +150,98 @@ class EditDeployment extends Component {
     });
   }
 
-  render() {
-    if (this.state.deploymentUpdated) {
-      return <Redirect to={`/deployments/${this.getDeploymentId()}`} />;
+  submitEditorContent() {
+    let parsedDeployment = undefined;
+    try {
+      parsedDeployment = JSON.parse(this.state.editorDeployment, null, 2);
+    } catch (syntaxError) {
+      this.setState({
+        deploymentUpdated: false,
+        errorMessage: syntaxError.message,
+      });
     }
 
-    const deployment = this.state.deployment;
+    if (parsedDeployment !== undefined) {
+      this.props.updateDeployment(parsedDeployment).then(() => {
```

Comment:
This function expects [two parameters](https://github.com/DataCater/datacater/blob/main/ui/src/actions/deployments.js#L112).